Mathlib/Order/LiminfLimsup.lean

```diff
@@ -0,0 +1,1242 @@
+  simp only [blimsup_eq_limsup_subtype, Function.comp,
+    (atTop_basis.comap ((↑) : { x | p x } → ℕ)).limsup_eq_infᵢ_supᵢ, supᵢ_subtype, supᵢ_and]
+  simp only [mem_setOf_eq, mem_preimage, mem_Ici, not_le, infᵢ_pos]
+#align filter.blimsup_eq_infi_bsupr_of_nat Filter.blimsup_eq_infᵢ_bsupᵢ_of_nat
+
+/-- In a complete lattice, the liminf of a function is the infimum over sets `s` in the filter
+of the supremum of the function over `s` -/
+theorem liminf_eq_supᵢ_infᵢ {f : Filter β} {u : β → α} : liminf u f = ⨆ s ∈ f, ⨅ a ∈ s, u a :=
+  @limsup_eq_infᵢ_supᵢ αᵒᵈ β _ _ _
+#align filter.liminf_eq_supr_infi Filter.liminf_eq_supᵢ_infᵢ
+
+theorem liminf_eq_supᵢ_infᵢ_of_nat {u : ℕ → α} : liminf u atTop = ⨆ n : ℕ, ⨅ i ≥ n, u i :=
+  @limsup_eq_infᵢ_supᵢ_of_nat αᵒᵈ _ u
+#align filter.liminf_eq_supr_infi_of_nat Filter.liminf_eq_supᵢ_infᵢ_of_nat
+
+theorem liminf_eq_supᵢ_infᵢ_of_nat' {u : ℕ → α} : liminf u atTop = ⨆ n : ℕ, ⨅ i : ℕ, u (i + n) :=
+  @limsup_eq_infᵢ_supᵢ_of_nat' αᵒᵈ _ _
+#align filter.liminf_eq_supr_infi_of_nat' Filter.liminf_eq_supᵢ_infᵢ_of_nat'
+
+theorem HasBasis.liminf_eq_supᵢ_infᵢ {p : ι → Prop} {s : ι → Set β} {f : Filter β} {u : β → α}
+    (h : f.HasBasis p s) : liminf u f = ⨆ (i) (_hi : p i), ⨅ a ∈ s i, u a :=
+  @HasBasis.limsup_eq_infᵢ_supᵢ αᵒᵈ _ _ _ _ _ _ _ h
+#align filter.has_basis.liminf_eq_supr_infi Filter.HasBasis.liminf_eq_supᵢ_infᵢ
+
+theorem bliminf_eq_supᵢ_binfᵢ {f : Filter β} {p : β → Prop} {u : β → α} :
+    bliminf u f p = ⨆ s ∈ f, ⨅ (b) (_hb : p b ∧ b ∈ s), u b :=
+  @blimsup_eq_infᵢ_bsupᵢ αᵒᵈ β _ f p u
+#align filter.bliminf_eq_supr_binfi Filter.bliminf_eq_supᵢ_binfᵢ
+
+theorem bliminf_eq_supᵢ_binfᵢ_of_nat {p : ℕ → Prop} {u : ℕ → α} :
+    bliminf u atTop p = ⨆ i, ⨅ (j) (_hj : p j ∧ i ≤ j), u j :=
+  @blimsup_eq_infᵢ_bsupᵢ_of_nat αᵒᵈ _ p u
+#align filter.bliminf_eq_supr_binfi_of_nat Filter.bliminf_eq_supᵢ_binfᵢ_of_nat
+
+theorem limsup_eq_infₛ_supₛ {ι R : Type _} (F : Filter ι) [CompleteLattice R] (a : ι → R) :
+    limsup a F = infₛ ((fun I => supₛ (a '' I)) '' F.sets) := by
+  refine' le_antisymm _ _
+  · rw [limsup_eq]
+    refine' infₛ_le_infₛ fun x hx => _
+    rcases(mem_image _ F.sets x).mp hx with ⟨I, ⟨I_mem_F, hI⟩⟩
+    filter_upwards [I_mem_F]with i hi
+    exact hI ▸ le_supₛ (mem_image_of_mem _ hi)
+  · refine'
+      le_infₛ_iff.mpr fun b hb =>
+        infₛ_le_of_le (mem_image_of_mem _ <| Filter.mem_sets.mpr hb) <| supₛ_le _
+    rintro _ ⟨_, h, rfl⟩
+    exact h
+set_option linter.uppercaseLean3 false in
+#align filter.limsup_eq_Inf_Sup Filter.limsup_eq_infₛ_supₛ
+
+theorem liminf_eq_supₛ_infₛ {ι R : Type _} (F : Filter ι) [CompleteLattice R] (a : ι → R) :
+    liminf a F = supₛ ((fun I => infₛ (a '' I)) '' F.sets) :=
+  @Filter.limsup_eq_infₛ_supₛ ι (OrderDual R) _ _ a
+set_option linter.uppercaseLean3 false in
+#align filter.liminf_eq_Sup_Inf Filter.liminf_eq_supₛ_infₛ
+
+-- Porting note: simp_nf linter, Left-hand side does not simplify when using simp on itself.
+-- @[simp]
```

Comment:
Did you test whether the linter is actually right? If it's claim is false, please add a nolint (and leave a porting note that mentions why).

Reply:
I had too much trust in the linter, it is indeed incorrect here. Thanks!